src/lib/mod.rs

```diff
@@ -12,46 +14,46 @@ const ESC_KEYSYM: u32 = 0xff1b;
 /// Then we grab on every possible combination of these masks by iterating
 /// through all the integers 0 to 255. This allows us to grab Esc, Shift+Esc,
 /// CapsLock+Shift+Esc, or any other combination.
-const KEY_GRAB_MASK_MAX: xcb::ModMask = (xcb::MOD_MASK_5 * 2) - 1;
+const KEY_GRAB_MASK_MAX: u16 = (xproto::ModMask::M5 as u16 * 2) - 1;
 
 #[derive(Clone, Copy)]
 pub struct HacksawResult {
     pub window: u32,
-    pub rect: xcb::Rectangle,
+    pub rect: xproto::Rectangle,
 }
 
 impl HacksawResult {
     pub fn x(&self) -> i16 {
-        self.rect.x()
+        self.rect.x
     }
     pub fn y(&self) -> i16 {
-        self.rect.y()
+        self.rect.y
     }
     pub fn width(&self) -> u16 {
-        self.rect.width()
+        self.rect.width
     }
     pub fn height(&self) -> u16 {
-        self.rect.height()
+        self.rect.height
     }
 
     pub fn relative_to(&self, parent: HacksawResult) -> HacksawResult {
         HacksawResult {
             window: self.window,
-            rect: xcb::Rectangle::new(
-                parent.x() + self.x(),
-                parent.y() + self.y(),
-                self.width(),
-                self.height(),
-            ),
+            rect: xproto::Rectangle {
+                x: parent.x() - self.x(),
+                y: parent.y() - self.y(),
```

Comment:
```suggestion
                x: parent.x() + self.x(),
                y: parent.y() + self.y(),
```

Why did I change this to - instead of +????